servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.cpp

```diff
@@ -1262,6 +1265,23 @@ void RenderForwardClustered::_fill_render_list(RenderListType p_render_list, con
 					inst->gi_offset_cache = 0xFFFFFFFF;
 				}
 			}
+
+#ifdef DEBUG_ENABLED
+			if (unlikely(get_debug_draw_mode() == RS::VIEWPORT_DEBUG_DRAW_UV2_TEXEL_DENSITY)) {
+				if (inst->data->base_type == RS::INSTANCE_MESH && inst->data->use_baked_light) {
+					Size2 lightmap_size = mesh_storage->mesh_get_lightmap_size_hint(inst->data->base) * inst->data->lightmap_scale;
+
+					if (inst->shader_parameters_offset == -1) {
```

Comment:
```suggestion
					if (inst->shader_parameters_offset == -1 && !inst->data->instance_allocated_shader_parameters && ) {
```

In my test project I get a spammed error because ``global_variables_instance_allocate()`` return ``-1`` if the instance has already been allocated. Adding the check for ``instance_allocated_shader_parameters`` ensures that this doesn't happen.

Reply:
I've added appropriate checks to mitigate the spammed error.

Reply:
Did you test just adding ``&& !inst->data->instance_allocated_shader_parameters``? From my test it was able to remove the spammed error and it is much simpler code